model/src/main/java/org/cbioportal/model/summary/MutationSummary.java

```diff
@@ -0,0 +1,259 @@
+package org.cbioportal.model.summary;
+
+import java.io.Serializable;
+
+public abstract class MutationSummary implements Serializable {
```

Comment:
I think a fundamental problem we have is that we currently have only a single model which is used both for the persistence layer transactions and also for the service/presentation layer functions. If we put too much domain knowledge into the model classes, there are inappropriate / distracting functionality there when using the model classes during development of the persistence layer. For mocking and unit testing, I think it is best to keep the lower level behaviors (if any) separate from the higher level behaviors. That way functionality can be easily tested in isolation. Also, embedded objects may be more appropriate to model at the higher level (perhaps a model of a genetic profile would contain mapping of mutation objects for example) but such an embedded collection might not be desired at the persistence level. If the added behaviors (more complex ones than getters and setters) have assumptions about which processing level will be calling them and what required initialization is needed, it is possible that these functions will be called prematurely.